### PR TITLE
Add tunnel_csum attribute to neutron barclamp

### PR DESCRIPTION
**Pull request by Itxaka**
As per http://openvswitch.org/pipermail/dev/2015-August/059335.html enabling tunnel_csum on the openvswitch agents may boost the network performance by a huge margin.

### CODE REVIEW

File: chef/cookbooks/neutron/templates/default/openvswitch_agent.ini.erb [hunk 1]
```diff
@@ -186,7 +186,9 @@ enable_distributed_routing = True
 # carrying GRE/VXLAN tunnel. The default value is False.
 #
 # tunnel_csum = False
-
+<% if @tunnel_csum -%>
+tunnel_csum = True
+<% end -%>
```
Comment (aplanas) [inline chef/cookbooks/neutron/templates/default/openvswitch_agent.ini.erb]: Can we have the removed new line here?

**Comment by aplanas** [comment]
LGTM +1

**Comment by Itxaka** [comment]
Btw, this only adds the option, and sets the default value to false. Unfortunately I could not find a lot of information regarding what could be affected by setting it to true by default.

**Comment by toabctl** [comment]
you need to update the schema-revision in the template-neutron.json file

**Comment by bmwiedemann** [comment]
Enabling offloading to network devices has often broken network in subtle ways, because much of the offloading-code in the various drivers did not handle the extra 4 byte added by tagged VLANs correctly. And apparently very few people use tagged VLANs.
So it is good that it is off by default.

File: chef/data_bags/crowbar/template-neutron.json [hunk 2]
```diff
@@ -118,7 +118,7 @@
     "neutron": {
       "crowbar-revision": 0,
       "crowbar-applied": false,
-      "schema-revision": 48,
+      "schema-revision": 49,
```
Comment (toabctl) [inline chef/data_bags/crowbar/template-neutron.json]: Sorry for not mentioning it - we agreed to start with 100 for Cloud7 :)

**Comment by Itxaka** [comment]
I also bumped the name of the migration. Should they go together? I actually have no idea how the migration is applied and based on what, but it makes sense that would run the migration up tot he schema number?

**Comment by vuntz** [comment]
Can you squash all commits into one commit? (you can force push, to not have to reopen another PR) After that, +1.

**Comment by nkrinner** [comment]
@Itxaka As far as I know the migration number has to match the schema revision. Thanks!

**Comment by toabctl** [comment]
> I actually have no idea how the migration is applied and based on what, but it makes sense that would run the migration up tot he schema number?

you can have a look at the rake tasks on the admin node: 
```
cd /opt/dell/crowbar_framework/bin/
./rake  -T|grep migrate
```

To get an overview:
```
RAILS_ENV=production rake crowbar:schema_migrate_status
```

To migrate all barclamps:
```
RAILS_ENV=production rake crowbar:schema_migrate
```

**Comment by toabctl** [comment]
+1

**Comment by rsalevsky** [comment]
+1

**Comment by sayalilunkad** [comment]
+1

File: chef/data_bags/crowbar/template-neutron.json [hunk 3]
```diff
@@ -13,6 +13,7 @@
       "dhcp_domain": "openstack.local",
       "use_lbaas": true,
       "use_dvr": false,
+      "tunnel_csum": false,
```
Comment (vuntz) [inline chef/data_bags/crowbar/template-neutron.json]: Could we put the attribute inside a ovs attribute, to make it clear that it's ovs-specific?

Comment (Itxaka) [inline chef/data_bags/crowbar/template-neutron.json]: done

File: chef/data_bags/crowbar/migrate/neutron/100_add_tunnel_csum.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+def upgrade(ta, td, a, d)
+  unless a.key?("ovs")
+    a["ovs"] = {}
+  end
```
Comment (vuntz) [inline chef/data_bags/crowbar/migrate/neutron/100_add_tunnel_csum.rb]: `a["ovs"] ||= {}`

**Comment by vuntz** [comment]
Thanks for the change. There's one minor code style comment from me, but generally +1.

Btw, you can squash all commits together.

File: chef/data_bags/crowbar/migrate/neutron/100_add_tunnel_csum.rb (hunk 4 above)
Comment (Itxaka) [inline chef/data_bags/crowbar/migrate/neutron/100_add_tunnel_csum.rb]: done!

**Comment by vuntz** [comment]
+1

**Comment by Itxaka** [comment]
@aspiers @vuntz @nkrinner @toabctl I have the possibility of merging this myself, but not sure if there is some key members that are in charge of merging stuff or  all team members can just merge any patches (as long as everything is green!)

**Comment by vuntz** [comment]
@Itxaka you can merge your PR yourself, it's fine. The only policy is that it should not be merged unless there are two +1 for the latest status of your PR (here, there's only one). So I'd recommend asking one of the persons who gave an earlier +1 to confirm their +1 :-)

**Comment by toabctl** [comment]
+1

**Pull request merged**